### DISCUSSION TITLE
Custom Error Response Body

### DISCUSSION THREAD

**Discussion by indra-an** [apollographql/router#1706] [Q&A]
Is there any way to custom/add key to response body with rhai script or rs plugin?
Because I need it to be compatible with old apps

for example
`{
    "status": 403,
    "message": "Invalid or expired access token",
    "errors": [
{
            "message": "HTTP fetch failed from 'service': subgraph HTTP status error '403 Forbidden': Invalid or expired access token",
            "locations": [],
            "path": [],
            "extensions": {
                "type": "SubrequestHttpError",
                "service": "service",
                "reason": "HTTP fetch failed from 'service': subgraph HTTP status error '403 Forbidden': Invalid or expired access token"
            }
        }],
    "data": null
}`

**Comment by garypen** [comment]
Yes. You can do both of those things from `rhai` or via a `rust` plugin. We have [a number of examples](https://github.com/apollographql/router/tree/main/examples) that do just this.

For a broader view of our customization capabilities, [there is documentation](https://www.apollographql.com/docs/router/customizations/overview).
